video/out/opengl/video_shaders.c

```diff
@@ -872,3 +872,22 @@ void pass_sample_unsharp(struct gl_shader_cache *sc, float param) {
     GLSLF("color = p + t * %f;\n", param);
     GLSLF("}\n");
 }
+
+void pass_sample_un360(struct gl_shader_cache *sc) {
+    GLSLF("{\n");
+    gl_sc_uniform_f(sc, "fov", M_PI/2);
+    gl_sc_uniform_f(sc, "theta0", M_PI);
+    gl_sc_uniform_f(sc, "phi0", M_PI/2);
+
+    GLSL(const float focal_x = .5/tan(fov/2);)
+    GLSL(const float focal_y = .5/tan(fov/4);)
+    GLSL(float theta = atan(HOOKED_pos.x-.5, focal_x);)
+    GLSL(float phi = atan(HOOKED_pos.y-.5, focal_y);)
+    GLSLF("float x = (theta+theta0)/%f;\n", 2*M_PI);
+    GLSL(if (x < 0 || x > 1) x -= floor(x);)
```

Comment:
If you don't care about `x==1` case, use `fract()`? Even if you do, make `x==1` a special case seems more reasonable to me.